hazelcast/src/test/java/com/hazelcast/map/impl/operation/MapFetchIndexOperationTest.java

```diff
@@ -0,0 +1,305 @@
+
+        MapFetchIndexOperationResult result = operationService.createInvocationBuilder(
+                MapService.SERVICE_NAME, operation, address).<MapFetchIndexOperationResult>invoke().get();
+
+
+        assertResultSorted(result, Arrays.asList(
+                new Person("person2", 39),
+                new Person("person3", 60)
+        ));
+    }
+
+    @Test
+    public void whenSizeHintIsSmall_thenFetchInMultipleOperations() throws ExecutionException, InterruptedException {
+        PartitionIdSet partitions = getLocalPartitions(instance1);
+
+        IndexIterationPointer[] pointers = new IndexIterationPointer[2];
+        pointers[0] = new IndexIterationPointer(30, true, 40, true, false);
+        pointers[1] = new IndexIterationPointer(50, true, 60, true, false);
+
+        MapOperationProvider operationProvider = getOperationProvider();
+        MapOperation operation = operationProvider.createFetchIndexOperation(mapName, orderedIndexName, pointers, partitions, 1);
+
+        Address address = instance1.getCluster().getLocalMember().getAddress();
+        OperationServiceImpl operationService = getOperationService(instance1);
+
+        MapFetchIndexOperationResult result = operationService.createInvocationBuilder(
+                MapService.SERVICE_NAME, operation, address).<MapFetchIndexOperationResult>invoke().get();
+
+
+        assertResultSorted(result, Arrays.asList(new Person("person2", 39)));
+
+        assertTrue(result.getContinuationPointers()[0].isDone());
+        assertFalse(result.getContinuationPointers()[1].isDone());
+
+        operation = operationProvider.createFetchIndexOperation(
+                mapName, orderedIndexName, result.getContinuationPointers(), partitions, 4);
+
+        result = operationService.createInvocationBuilder(
+                MapService.SERVICE_NAME, operation, address).<MapFetchIndexOperationResult>invoke().get();
+
+        assertResultSorted(result, Arrays.asList(new Person("person3", 60)));
+
+        assertTrue(result.getContinuationPointers()[0].isDone());
+        assertTrue(result.getContinuationPointers()[1].isDone());
+    }
+
+    @Test
+    public void whenMultipleObjectsHasSameValue_thenOperationCanReturnMoreThanSizeHint() throws ExecutionException, InterruptedException {
+        PartitionIdSet partitions = getLocalPartitions(instance1);
+
+        IndexIterationPointer[] pointers = new IndexIterationPointer[1];
+        pointers[0] = new IndexIterationPointer(30, true, 60, true, false);
+
+        MapOperationProvider operationProvider = getOperationProvider();
+        MapOperation operation = operationProvider.createFetchIndexOperation(mapName, orderedIndexName, pointers, partitions, 3);
+
+        Address address = instance1.getCluster().getLocalMember().getAddress();
+        OperationServiceImpl operationService = getOperationService(instance1);
+
+        MapFetchIndexOperationResult result = operationService.createInvocationBuilder(
```

Comment:
Is this operation supposed to run with [`invokeOnPartition(Operation op)`](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/OperationService.java#L129) method?